contracts/0.4.24/nos/NodeOperatorsRegistry.sol

```diff
@@ -577,10 +577,10 @@ contract NodeOperatorsRegistry is AragonApp, Versioned {
 
             // upd totals
             Packed64x4.Packed memory summarySigningKeysStats = _loadSummarySigningKeysStats();
-            summarySigningKeysStats.set(
-                SUMMARY_EXITED_KEYS_COUNT_OFFSET,
-                uint64(int64(summarySigningKeysStats.get(SUMMARY_EXITED_KEYS_COUNT_OFFSET)) + totalExitedValidatorsDelta)
-            );
+            // no need to check SUMMARY_EXITED_KEYS_COUNT <= UIN64_MAX due to it is the sum of
+            // operators's TOTAL_EXITED_KEYS_COUNT, each of which does not exceed operator's TOTAL_DEPOSITED_KEYS_COUNT,
+            // which in turn is limited by SafeMath64 when calculating SUMMARY_DEPOSITED_KEYS_COUNT
+            summarySigningKeysStats.set(SUMMARY_EXITED_KEYS_COUNT_OFFSET, uint64(uint256(int256(summarySigningKeysStats.get(SUMMARY_EXITED_KEYS_COUNT_OFFSET)) + totalExitedValidatorsDelta)));
```

Comment:
As a rough idea to get rid of chain casts: we can add a method to the `Packed64x4` lib, which will accept delta as `int128` or `int256` and apply it to value. Then we can write something like: `summarySigningKeysStats.slide(SUMMARY_EXITED_KEYS_COUNT_OFFSET, 
 totalExitedValidatorsDelta)`.

Reply:
yep, but the basic idea is that all numbers we have are uint64 values, so when we convert them to int64, the math can break if the absolute values are higher than uin64_max/2.

to avoid this, we just use a cast to higher bits numbers to do all the math. But again, no one result value will exceed uint64. So converting back to uint64 is safe. Thus, theoretically, even uint65 could be enough)